### ISSUE TITLE
Syncing prometheus fails when a ns error occurs but none of other resources have this problem

### ISSUE THREAD

**Issue by lilic** [prometheus-operator/prometheus-operator#3347]
**What happened?**

Passed two namespaces that do not exist and two that do exist via the prometheus operator flag `--namespaces`. Seen the expected errors of:
> E0720 10:40:22.043151   18190 reflector.go:178] github.com/coreos/prometheus-operator/pkg/prometheus/operator.go:484: Failed to list *v1.Namespace: namespaces "blah" not found

All resources were up and running and syncing for example:
> level=info ts=2020-07-20T08:40:33.573732Z caller=operator.go:427 component=alertmanageroperator msg="sync alertmanager" key=monitoring/main

and it correctly picked up the Alertmanager CR instance.

But it failed to the same for prometheus CR instance and create the needed resources, and in the logs it seems like the syncing never actually happened.

**Did you expect to see something different?**

When only the existing namespaces are created this error does not happen, I can see the sync of prometheus in the logs and prometheus resources are created correctly:
> level=info ts=2020-07-20T08:37:02.519745Z caller=operator.go:171 component=alertmanageroperator msg="successfully synced all caches"
level=info ts=2020-07-20T08:37:02.519754Z caller=operator.go:407 component=prometheusoperator msg="successfully synced all caches"
level=info ts=2020-07-20T08:37:02.519874Z caller=operator.go:427 component=alertmanageroperator msg="sync alertmanager" key=monitoring/main
level=info ts=2020-07-20T08:37:02.519917Z caller=operator.go:1122 component=prometheusoperator msg="sync prometheus" key=monitoring/k8s
level=info ts=2020-07-20T08:37:02.555934Z caller=operator.go:427 component=alertmanageroperator msg="sync alertmanager" key=monitoring/main
level=info ts=2020-07-20T08:37:02.620839Z caller=operator.go:1122 component=prometheusoperator msg="sync prometheus" key=monitoring/k8s

I expected that prometheus operator ignores the errors and tolerates those and keeps reconciles despite any non related errors like namespaces in the above example.

**How to reproduce it (as minimally and precisely as possible)**:
Start prometheus operator with three non existing namespaces, example:
`--namespaces=default,ns2,blah,ns1,monitoring`

**Environment**

* Prometheus Operator version:

Tried with 0.40 and master

* Kubernetes version information:

    `kubectl version`
    

* Kubernetes cluster kind:

1.18.3 vanilla minikube cluster.

**Comment by brancz** [comment]
This should be fixed by https://github.com/coreos/prometheus-operator/issues/3218

**Comment by lilic** [comment]
tl;dr of the above issue is that we should consider being in general tolerant of errors and failures more as these things are very hard to debug for users if we do not perform the sync and at the same time never actually kill the process, we just swallow those errors.
